include/aspect/geometry_model/box.h

```diff
@@ -239,6 +253,12 @@ namespace aspect
          * A pointer to the initial topography model.
          */
         InitialTopographyModel::Interface<dim> *topo_model;
+
+        /**
+         * Variables to hold surface topography when checking the depth including mesh deformation.
+         */
+        std::vector<double> surface_x;
+        std::vector<double> surface_y;
```

Comment:
The documentation here suggests that you only need these variables during specific phases of the algorithm. In that case, you could make them local to the function that needs them.

If you need these variables to communicate between functions (e.g., `update()` computes the values, `depth_including_mesh_deformation()` uses them), then member variables are the right way to go. But you should find a dimension-independent way to deal with this. As already mentioned, you could use one of the deal.II tables if the data is structured on an x-y grid.

Reply:
I've rolled all the x,y and z values that may be at the surface into a local vector of vectors, and now have the Interpolated function as the member variable. Does it make sense to do it this way?